gaia-fast-list.js

```diff
@@ -0,0 +1,383 @@
+function Configuration(config) {
+  Object.assign(this, config);
+  this.container = config.container;
+  this.model = config.model || [];
+  debug('config init', this);
+}
+
+Configuration.prototype = {
+  setModel: function(model) {
+    this.model = model;
+    this.sections = this.sectionize(this.model);
+  },
+
+  sectionize: function(data) {
+    var hash = {};
+
+    data.forEach(item => {
+      var section = this.getSectionName(item);
+      if (!hash[section]) { hash[section] = []; }
+      hash[section].push(item);
+    });
+
+    return hash;
+  },
+
+  sectionTemplate: '<section><h2> </h2><div class="background"></div></section>',
+  itemTemplate: '<li><a><div><h3> </h3><p> </p></div><img/></a></li>',
+
+  populateItem: function(el, i) {
+    debug('populate item');
+    var data = this.getRecordAt(i);
+    var els = {};
+
+    els.link = el.firstChild;
+    els.div = els.link.firstChild;
+    els.title = els.div.firstChild;
+    els.body = els.title.nextSibling;
+    els.image = els.div.nextSibling;
+
+    var title = getProp(data, this.itemKeys.title);
+    var image = getProp(data, this.itemKeys.image);
+    var link = getProp(data, this.itemKeys.link);
+    var body = getProp(data, this.itemKeys.body);
+
+    els.link.href = link;
+    els.title.firstChild.data = title;
+    els.body.firstChild.data = body;
+
+    if (image) {
+      els.image.style.display = '';
+      els.image.src = image;
+    } else {
+      els.image.style.display = 'none';
+    }
+  },
+
+  itemKeys: {
+    title: 'title',
+    body: 'body',
+    image: 'image',
+    link: 'link'
+  },
+
+  populateSection: function(el, section) {
+    var title = el.firstChild;
+    var height = this.fullSectionHeight(section);
+    var background = title.nextSibling;
+
+    background.style.height = height + 'px';
+    title.firstChild.data = section;
+  },
+
+  getSections: function() {
+    return Object.keys(this.sections || {});
+  },
+
+  sectionHeaderHeight: function() {
+    return this._headerHeight;
+  },
+
+  fullSectionHeight: function(key) {
+    return this.sections[key].length * this.itemHeight();
+  },
+
+  fullSectionLength: function(key) {
+    return this.sections[key].length;
+  },
+
+  getRecordAt: function(index) {
+    return this.model[index];
+  },
+
+  getSectionFor: function(index) {
+    var item = this.getRecordAt(index);
+    return this.getSectionName(item);
+  },
+
+  indexAtPosition: function(pos) {
+    debug('index at position', pos);
+    var headerHeight = this.sectionHeaderHeight();
+    var itemHeight = this.itemHeight();
+    var index = 0;
+    var items;
+
+    for (var key in this.sections) {
+      items = this.sections[key];
+
+      pos -= headerHeight;
+      var sectionHeight = items.length * itemHeight;
+
+      if (pos > sectionHeight) {
+        pos -= sectionHeight;
+        index += items.length;
+        continue;
+      }
+
+      for (var i = 0; i < items.length; i++) {
+        pos -= itemHeight;
+        index++;
```

Comment:
`indexAtPosition(0)` will return `1` here, causing the first element flcker.
You need to put the `index++` after giving the function a chance to `return index;'`